passengerfeedback/passenger_client/main.go

```diff
@@ -0,0 +1,249 @@
+		return
+	}
+	limit, err := strconv.Atoi(strLimit)
+	if err != nil {
+		c.JSON(http.StatusOK, gin.H{"errorCode": pb.Error_FAIL, "message": "Limit must be a number"})
+		return
+	}
+
+	response, err := getFeedbackByPassengerId(int32(passengerId), int32(offset), int32(limit))
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"errorCode": pb.Error_FAIL, "message": err.Error()})
+		log.Print(err)
+		return
+	}
+	c.JSON(http.StatusOK, gin.H{"errorCode": response.ErrorCode, "message": response.Msg, "data": response.Data})
+}
+
+func handleGetFeedbackByBookingCode(c *gin.Context) {
+	strBookingCode, b := c.GetQuery("bookingCode")
+	if !b || len(strings.TrimSpace(strBookingCode)) == 0 {
+		c.JSON(http.StatusOK, gin.H{"errorCode": pb.Error_FAIL, "message": "Missing bookingCode"})
+		return
+	}
+	strOffset, b := c.GetQuery("offset")
+	if !b || len(strings.TrimSpace(strOffset)) == 0 {
+		c.JSON(http.StatusOK, gin.H{"errorCode": pb.Error_FAIL, "message": "Missing Offset"})
+		return
+	}
+	offset, err := strconv.Atoi(strOffset)
+	if err != nil {
+		c.JSON(http.StatusOK, gin.H{"errorCode": pb.Error_FAIL, "message": "Offset must be a number"})
+		return
+	}
+	strLimit, b := c.GetQuery("limit")
+	if !b || len(strings.TrimSpace(strLimit)) == 0 {
+		c.JSON(http.StatusOK, gin.H{"errorCode": pb.Error_FAIL, "message": "Missing Limit"})
+		return
+	}
+	limit, err := strconv.Atoi(strLimit)
+	if err != nil {
+		c.JSON(http.StatusOK, gin.H{"errorCode": pb.Error_FAIL, "message": "Limit must be a number"})
+		return
+	}
+
+	response, err := getFeedbackByBookingCode(strBookingCode, int32(offset), int32(limit))
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"errorCode": pb.Error_FAIL, "message": err.Error()})
+		log.Print(err)
+		return
+	}
+	c.JSON(http.StatusOK, gin.H{"errorCode": response.ErrorCode, "message": response.Msg, "data": response.Data})
+}
+
+func handleDeleteFeedbackByPassengerId(c *gin.Context) {
+	var argument struct {
+		PassengerId string
+	}
+	err := c.BindJSON(&argument)
+	if err != nil {
+		c.String(400, "invalid param \n")
+		log.Print(err)
+		return
+	}
+	if len(strings.TrimSpace(argument.PassengerId)) == 0 {
+		c.JSON(http.StatusOK, gin.H{"errorCode": pb.Error_FAIL, "message": "Missing passengerId"})
+		return
+	}
+	passengerId, err := strconv.Atoi(argument.PassengerId)
+	if err != nil {
+		c.JSON(http.StatusOK, gin.H{"errorCode": pb.Error_FAIL, "message": "PassengerId must be a number"})
+		return
+	}
+
+	response, err := deleteFeedbackByPassengerId(int32(passengerId))
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"errorCode": pb.Error_FAIL, "message": err.Error()})
+		log.Print(err)
+		return
+	}
+	c.JSON(http.StatusOK, gin.H{"errorCode": response.ErrorCode, "message": response.Msg})
+}
+
+func addPassengerFeedback(bookingCode string, passengerId int32, feedback string) (res *pb.PassengerFeedbackResponse, err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
```

Comment:
using 2 different styles of returning (named return &  isn't a good practice, please pick one and stick to it.
`func addPassengerFeedback(bookingCode string, passengerId int32, feedback string) (res *pb.PassengerFeedbackResponse, err error) {`
`func getFeedbackByPassengerId(passengerId, offset, limit int32) (*pb.PassengerFeedbackSliceResponse, error) {`